sys/include/evtimer_msg_mbox.h

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2020 Simon Brummer <simon.brummer@posteo.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @addtogroup  sys_evtimer
+ * @{
+ *
+ * @file
+ * @brief       IPC Message based evtimer event, extended with mbox support.
+ *
+ * @author      Simon Brummer <simon.brummer@posteo.de>
+ */
+#ifndef EVTIMER_MSG_MBOX_H
+#define EVTIMER_MSG_MBOX_H
+
+#include "assert.h"
+#include "msg.h"
+#include "mbox.h"
+#include "evtimer.h"
+#include "evtimer_msg.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   IPC-message event with optional mbox support.
+ * @extends evtimer_msg_event_t
+ */
+typedef struct {
+    evtimer_event_t event;   /**< event base class */
+    msg_t msg;               /**< IPC message so send on event */
+    mbox_t *target_mbox;     /**< mbox the IPC message can be stored */
+} evtimer_msg_mbox_event_t;
+
+/**
+ * @brief   Initialze evtimer event.
+ *
+ * @param[in,out] event  Event to initialize.
+ */
+static inline void evtimer_msg_mbox_event_init(evtimer_msg_mbox_event_t *event)
+{
+    assert(event);
+    event->target_mbox = NULL;
+}
+
+/**
+ * @brief   Adds event to an event timer that stores an IPC message in a mbox.
+ *
+ * @param[in] evtimer       Timer to add @p event.
+ * @param[in] event         Event with mbox support.
+ * @param[in] target_mbox   mbox to store IPC message.
+ */
+static inline void evtimer_add_msg_mbox(evtimer_t *evtimer, evtimer_msg_mbox_event_t *event,
+                                        mbox_t *target_mbox)
+{
+    assert(evtimer);
+    assert(event);
+    assert(target_mbox);
+    event->target_mbox = target_mbox;
+    evtimer_add(evtimer, &(event->event));
+}
+
+/**
+ * @brief   Event handler for IPC messages with mbox support.
+ *
+ * @param[in] event   The event to handle
+ */
+static inline void _evtimer_msg_mbox_handler(evtimer_event_t *event)
+{
+    assert(event);
+    evtimer_msg_mbox_event_t *mbevent = (evtimer_msg_mbox_event_t *)event;
+
+    /* If a message box is set, store message, otherwise
+     * fallback to message handler of evtimer_msg */
```

Comment:
Why? This makes things more complicated than they need to. If you want to send to a thread's message queue rather than an `mbox` use `evtimer_msg` and not use implicit behavior.

Reply:
I'll see your point there. Originally this PR was intended to be able to send messages to mboxes and normal messages from the same evtimer, reducing the nessecary changes in gnrc_tcp later on. 

However rewriting this to only support mboxes is the better approach.